### PR TITLE
Remove sync/sign in language from avatar menu.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#2234

This fix modifies the current profile's button in the avatar popup menu.
It removes the sync overlay from the profile icon and replaces "Sync
disabled"/"Not signed in" labels with the profile's name.

Adds a browser test to verify that the button shows profile name.

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
**Automated test:**
`npm run test -- brave_browser_tests --filter=BraveProfileChooserViewTest.TestCurrentProfileView`

**Manual test:**
- Start Brave browser,
- Click on the avatar menu button in the toolbar,
- Verify that the first button in the menu shows profile avatar with no overlays and the text is the name of the profile.
- Click on the button - this should result in profile settings page showing up,
- In the profile settings page change the profile name,
- Click on the avatar menu button again,
- Verify that the profile button shows the new name.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/ui/views/profiles/brave_profile_chooser_view.h [hunk 1]
```diff
@@ -23,6 +23,10 @@ class BraveProfileChooserView : public ProfileChooserView {
 
   void ResetView() override;
 
+  views::View* CreateDiceSyncErrorView(const AvatarMenu::Item& avatar_item,
```
Comment (darkdh) [inline browser/ui/views/profiles/brave_profile_chooser_view.h]: shouldn't we override `CreateCurrentProfileView`?

Comment (mkarolin) [reply browser/ui/views/profiles/brave_profile_chooser_view.h]: We could, would have to account for Guest profiles as well then.

**Comment by kjozwiak** [comment]
@mkarolin approving uplift request to `0.58.x` after deliberating with @rebron & @srirambv 👍Please make sure that all the appropriate labels are removed/added and the corresponding issue is moved to the correct milestone.

**Comment by mkarolin** [comment]
0.58.x: 12882ce

**Pull request merged**